vulcan/lib/client/jsx/components/workflow/steps/step_icon.tsx

```diff
@@ -0,0 +1,62 @@
+import React, {useContext} from 'react';
+
+import {VulcanContext} from '../../../contexts/vulcan_context';
+import Icon from 'etna-js/components/icon';
+import AnimatedClock from './animated_clock';
+import {WorkflowStep} from '../../../api_types';
+import {labelOfStepOrGroupedStep, statusStringOfStepOrGroupedStep} from '../../../selectors/workflow_selectors';
+import {STATUS} from '../../../api_types';
+import {WorkflowStepGroup} from "../user_interactions/inputs/input_types";
+import {makeStyles} from '@material-ui/core/styles';
+import Tooltip from '@material-ui/core/Tooltip';
+import CheckIcon from '@material-ui/icons/Check';
+import ClockIcon from '@material-ui/icons/AccessTime';
+import ErrorIcon from '@material-ui/icons/HighlightOff';
+
+const useStyles = makeStyles((theme) => ({
+  icon: {
+    opacity: 0.5
+  },
+  [STATUS.COMPLETE]: {
+    color: 'green'
+  },
+  [STATUS.RUNNING]: {
+    color: 'gray'
+  },
+  [STATUS.PENDING]: {
+    color: 'gray'
+  },
+  [STATUS.ERROR]: {
+    color: 'red'
+  }
+}));
+
+type IconClass = typeof CheckIcon;
```

Comment:
I'm pretty sure this is the wrong way to do this, but :shrug: what the right way is.

Reply:
Perhaps the AnimatedClock should be wrapped in `SvgIcon` (https://v4.mui.com/api/svg-icon/) instead, and in which case the IconClass is just `typeof SvgIcon` ?